src/Analysis/Engine/Impl/DependencyResolution/PathResolverSnapshot.cs

```diff
@@ -228,19 +347,23 @@ public PathResolverSnapshot(PythonLanguageVersion pythonLanguageVersion)
             return true;
         }
 
-        private static Node CreateNewNodes(string modulePath, (int start, int length) unmatchedPathSpan, Node packageNode) {
-            if (packageNode != default) {
-                // Module name matches name of existing package        
-                return packageNode.ToModule(modulePath);
+        private static Node CreateNewNodes(in Edge lastEdge, in string modulePath, in (int start, int length) unmatchedPathSpan, out string fullModuleName) {
+            if (unmatchedPathSpan.length == 0) {
+                // Module name matches name of existing package
+                fullModuleName = GetFullModuleName(lastEdge);
+                return lastEdge.End.ToModule(modulePath, fullModuleName);
             }
 
             if (unmatchedPathSpan.start == GetModuleNameStart(modulePath)) {
                 // Module is added to existing package
-                return Node.CreateModule(modulePath.Substring(unmatchedPathSpan.start, unmatchedPathSpan.length), modulePath);
+                var name = modulePath.Substring(unmatchedPathSpan.start, unmatchedPathSpan.length);
+                fullModuleName = GetFullModuleName(lastEdge, name);
+                return Node.CreateModule(name, modulePath, fullModuleName);
             }
 
             var names = modulePath.Split(Path.DirectorySeparatorChar, unmatchedPathSpan.start, unmatchedPathSpan.length);
```

Comment:
Does it handle UNC ie `\\machine\share\...`?

Reply:
No. Do we have to support it?

Reply:
Doesn't work right now because it always uses `Path.DirectorySeparatorChar`, but it is easy to fix.